CODING_CONVENTIONS.md

```diff
@@ -0,0 +1,142 @@
+  ```java
+  MessageDigest md5 = MessageDigest.getInstance("MD5"); // BAD
+  // ... although this is a snippet from Skript source code. Legacy stuff!
+  ```
+* File or network IO, for now, should not be accessible from scripts
+  ```java
+  String fileName = file.getSingle(e); // Bad
+  ```
+* Code contributed must be licensed under GPLv3, by *you*
+  - Third party code under compatible licenses may be accepted in some cases
+
+And, additionally:
+* Everyone with direct push access must use two factor authentication
+  - https://xkcd.com/936/
+
+## Formatting
+* Tabs, no spaces (unless in code imported from other projects)
+** No tabs/spaces in empty lines
+* No trailing whitespace
+* At most 120 characters per line
+  - In Javadoc/multiline comments, at most 80 characters per line
+* When statements consume multiple lines, all lines but first have two tabs of additional indentation
+* Each class begins with an empty line
+* Separate method declarations with empty lines
+  - Empty line after last method in a class is *not* required
+  - Otherwise, empty line before and after method is a good rule of thumb
+* If fields have Javadoc, separate them with empty lines
+* Use empty lines liberally inside methods to improve readability
+* Use curly brackets to start and end most blocks
+  - Only when a conditional block (if or else) contains a single statement, they may be omitted
+  - When omitting brackets, still indent as if the code had brackets
+  - Avoid omitting brackets if it produces hard-to-read code
+* Annotations for methods and classes are placed in lines before their declarations, one per line
+* When there are multiple annotations, place them in order:
+  - @Override -> @Nullable -> @SuppressWarnings
+  - For other annotations, doesn't matter; let your IDE decide
+
+## Naming
+* Classes named in CapitalCamelCase
+* Fields and methods named in camelCase
+  - Final static fields may be named in SCREAMING_SNAKE_CASE
+* Localized messages should be named like m_message_name
+  - And that is the only place where snake_case is acceptable
+* Use prefixes where their use has been already estabilished (such as ExprSomeRandomThing)
+  - Otherwise, use postfixes (such as LoopSection)
+  
+## Comments
+* Prefer to comment *why* you're doing things instead of how you're doing them
+  - In case the code is particularly complex, though, do both!
+* Write all comments in readable English
+  - Remember the basic grammar rules
+  - Fun is allowed, if that does not hurt the readability
+* No matter if a comment contains a sentence or not, capitalize the first letter
+
+Your comments should look something like these:
+```java
+// A single line comment
+
+/**
+ * This is a Javadoc comment.
+ * @param first Description of the first parameter.
+ * @param second Description of the second parameter.
+ * @return Description of the return value.
+ */
+ 
+/*
+ * This is a block comment.
+ */
+```
```

Comment:
Looks good overall. Reminds me of tslint; in the spirit of common linting standards, there's room for a *bit* more clarification on a few things which are standard as linting rules:
- Do ALL functions need a Javadoc? Only public/protected ones?
- Are comments allowed at the end of a line? e.g. `int badExample = 5; // Used for something truly special`. My personal preference vote on that is no but either way it should be documented IMO.
- Is a space required after the comment denoter (`//comment` vs. `// comment`). Personal preference vote is yes but it should be documented.